fuse_models/include/fuse_models/unicycle_3d_state_cost_function.h

```diff
@@ -181,16 +181,17 @@ class Unicycle3DStateCostFunction : public ceres::SizedCostFunction<15, 3, 4, 3,
       acc_linear_pred[2],
       jacobians);
 
+    // Because the above function computes the jacobian wrt rpy orientation, it needs to be converted to
+    // quaternion orientation.
+    // This is still the case even although local parameterization is used because the jacobian matrix is the global
+    // size (so 15x4) and then later for optimization the jacobian internally gets updated with the jacobian of the
+    // transformation function from the global to the local size.
+    // See ceres 2.0.0: internal/ceres/residual_block.cc::143
+    // Also see: https://github.com/ceres-solver/ceres-solver/issues/387
     if (jacobians && jacobians[1])
     {
-      // Although local parameterization is used, the jacobian matrix is still the global size (so 15x4) and
-      // then later for optimization the jacobian internally gets updated with the jacobian of the transformation
-      // function from the global to the local size (see ceres 2.0.0: internal/ceres/residual_block.cc::143)
-      // Also see: https://github.com/ceres-solver/ceres-solver/issues/387
-
-      // Therefore we need to transform the roll, pitch, yaw orientation to quaternion space
-      // Note that this only works because jacobians[1] is originally 15x4 and the rpy jacobian
-      // is smaller (15x3) while vice versa a segfault might occur because of an out-of-bounds access
+      // Note that this only works without an out-of-bounds memory access because jacobians[1] is originally 15x4 and
+      // the rpy jacobian is smaller (15x3) while vice versa a segfault might occur because of an out-of-bounds access
       Eigen::Map<fuse_core::Matrix<double, 15, 3>> jacobian_orientation_rpy(jacobians[1]);
       Eigen::Map<fuse_core::Matrix<double, 15, 4>> jacobian_orientation_q(jacobians[1]);
```

Comment:
Well if jacobians is 15x4 -> it should have already the size it needs so the index would need to be 0

Reply:
I am sorry but I do not really understand your comment. Please clarify the comment further if needed.
In general, the comment in the code was mainly to make the reader aware that I intentionally take in the following two lines an `Eigen::Map` with different sizes but from the same array and I am aware that this could of course make problems but `jacobians[1]` (where `jacobians` is an array with arrays inside) is already 15x4 large array allocated by ceres itself so first using it as a 15x3 matrix is fine (without creating a new array) while if the case would be vice versa, this would be an issue.

Reply:
so jacobians[1] does acces the first element and you kick aout the last once for the 15x3 case?

Reply:
Also not the case but we discussed it now in person.
To have the misunderstanding possibly also resolved for future readers, I would like to highlight more the structure of the function parameter `jacobians`:
`jacobians` contains the jacobian matrices for all of the fuse/ceres variables meaning that for example in this case:
- `jacobians[0]` contains the full jacobian matrix (represented as flat array) for the position1 (3 dimensions with overall 15 residuals so `jacobians[0]` is a flat simple array with 45 entries which can be remapped using Eigen as 15x3 matrix)
- `jacobians[1]` contains the full jacobian matrix (represented as flat array) for the orientation1 (4 dimensions with overall 15 residuals so `jacobians[1]` is a flat simple array with 60 entries which can be remapped using Eigen as 15x4 matrix)
- `jacobians[2]` contains the full jacobian matrix (represented as flat array) for the linear_velocity1 (3 dimensions with overall 15 residuals so `jacobians[2]` is a flat simple array with 45 entries which can be remapped using Eigen as 15x3 matrix)
- etc.